Add Nutanix Integration

NouemanKHAL opened the pull request:
### What does this PR do?
<!-- A brief description of the change being made with this pull request. -->

 Adds Nutanix integration for monitoring Prism Central v4 infrastructure via Datadog Agent.

  ## Features

  - **Infrastructure Metrics**: Clusters, hosts, and VMs with stats (CPU, memory, storage, network)
  - **Activity Monitoring**: Events, audits, alerts, and tasks as Datadog events with a `ntnx_type` tag
  - **Category Tags**: Automatic tagging from Nutanix categories (with `ntnx_` prefix option)
  - **Resource Filtering**: Regex-based filtering for clusters, hosts, and VMs

  ## Technical Details

  - **API**: Nutanix Prism Central v4.0/v4.2 REST APIs
  - **Authentication**: Basic auth
  - **Rate Limiting**: Respects API rate limits with exponential backoff
  - **Testing**: Unit tests + Docker integration tests + AWS environment support

### Motivation
<!-- What inspired you to submit this pull request? -->
https://datadoghq.atlassian.net/browse/AI-5917

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Add the `qa/skip-qa` label if the PR doesn't need to be tested during QA.
- [ ] If you need to backport this PR to another branch, you can add the `backport/<branch-name>` label to the PR and it will automatically open a backport PR once this one is merged

drichards-87 commented:
Created DOCS-13544 for the editorial review.

sarah-witt commented on nutanix/datadog_checks/nutanix/infrastructure_monitor.py:
```diff
@@ -0,0 +1,821 @@
+            Number of VMs processed on this host
+        """
+        host_id = host.get("extId")
+        host_name = host.get("hostName")
```
Have you tested this with an agent running on the host?

NouemanKHAL replied:
That's currently a FR with Nutanix, the API does not expose the hostname, only the entity's name :/  
For now, the recommendation is to manually match them in the `datadog.yaml`, gonna add a note about it in the readme.

sarah-witt replied:
Sounds good!

NouemanKHAL replied:
Done!

NouemanKHAL commented:
Just a quick note for reviewers, below are the main changes of the last ~21 commits.
## New Features

  - Template rendering for event and audit messages, matching existing
  alert behavior.
  - Task events display affected alert titles by fetching and rendering
  referenced alerts.
  - In-memory activity caches enable cross-entity lookups and avoid
  redundant API calls.
  - Alerts collected before tasks so alert data is available for task
  enrichment.

  ## Tagging

  - Removed all `*_id` tags from infrastructure metrics and activity
  events (`ntnx_cluster_id`, `ntnx_host_id`, `ntnx_vm_id`,
  `ntnx_event_id`, `ntnx_alert_id`, `ntnx_audit_id`, `ntnx_task_id`,
  `ntnx_entity_id`, `ntnx_owner_id`, `ntnx_affected_entity_id`, etc.).

  ## Error Handling

  - Granular error isolation. Category, cluster, host, and VM failures are
   caught independently — a single failing entity no longer aborts the
  entire check run.

  ## Code Quality

  - Unified activity collection into a shared `_collect` / `_safe_collect`
   pattern, eliminating duplicated pagination, filtering, and cursor
  logic.
  - Extracted helpers (`_parse_iso`, `_add_source_entity_tags`,
  `_add_cluster_name_tag`) to remove repeated tag-building and timestamp
  parsing across process methods.
  - Consolidated stats submission, time-window calculation, and resource
  filter internals — reducing method count and indirection.

  ## Documentation

  - Added duplicate hostnames section explaining how to resolve hostname
  mismatches when the Agent runs inside a Nutanix VM.
  - Clarified single Agent is sufficient to monitor all Prism Central
  managed infrastructure.
  - Removed misleading `conf.yaml` comment about `powerState` filtering
  (default behavior unchanged).

sarah-witt commented on nutanix/tests/test_clusters.py:
```diff
@@ -0,0 +1,92 @@
+
+        raise ConnectionError("Connection failed")
+
+    mocker.patch('requests.Session.get', side_effect=mock_exception)
```
can we use the new way of mocking the http wrapper here?
